### ISSUE TITLE
Dragging from a kivy app to another or from kivy to the desktop.

### ISSUE THREAD

**Issue by encompass** [kivy/kivy#3557]
Is there a way to drag items (like an object to represent a file) from a Kivy app to another application or from Kivy app to the "desktop" of what ever the environment is?
This is a use case that would be commonly needed to make application integration better on the desktop environments and drag and drop is a very natural movement now a days.
This is a video showing the action we are looking for.
https://youtu.be/LZSYON4Z9fc

As per this request:
https://groups.google.com/forum/#!topic/kivy-users/QyjJdr4v3ho

**Comment by akshayaurora** [comment]
@encompass thanks for opening a formal issue, it's easier for us to track issues on github,  informal requests on the mailing request have the problem of going unnoticed due to the amount of foot fall there.

**Comment by encompass** [comment]
Totally know it.  I code too. :) Tell me if you need more help.

**Comment by sh3raawii** [comment]
I need that feature too, it's crucial for the UX of many apps.

**Comment by janssen** [comment]
This is really only half the solution.  It's also nice to be able to start the drag in Kivy, and have it drag from Kivy into an external object, which would receive an image or text or a filepath, depending on how the Kivy widget that originated the drag interpreted the data.

**Comment by David-mtl** [comment]
Hi, is there any news on this? Finding a cross-platform framework that allows dragging files outside of an app is much harder then I expected. Would love to use Kivy for it. As anyone found a way to do it?

**Comment by KeyWeeUsr** [comment]
I know it's not really a desirable solution, but... you can use `Clipboard` to get some values and manually paste them out of Kivy window. Other than that I think it depends on each platform's API - creating that fancy little arrow for drag&drop with each platform's API call.

In that video copying into browser pastes a path, which is put into Clipboard by nautilus(?). Copying text from browser into nautilus is the other way around i.e. browser puts into Clipboard `My Label ...` and in nautilus' window that kind of thing is converted into creating a `.txt` file, which basically means:

Kivy can provide:

* copying into its window (already here [Window.on_dropfile](https://kivy.org/docs/api-kivy.core.window.html#kivy.core.window.WindowBase.on_dropfile))
* copying from its window (basically calling `Clipboard.copy(<string>)`) and then some API call probably with [`Window.on_cursor_leave()`](https://kivy.org/docs/api-kivy.core.window.html#kivy.core.window.WindowBase.on_cursor_leave). Personally I don't really have a knowledge about API calls of other than Windows' and even there it's limited, however - just get some API calls into python and the events are already there.

Other stuff which you see in the video are handled by the other side of the two app interaction.

**Comment by David-mtl** [comment]
Yes it's not very desirable. What I'm looking for is a way to click something in my app and drag / drop into another app, a file explorer for example.

So if I understand correctly if I want to do it with Kivy I would need to: on Linux and OSX use python-xlib  and on Windows it would probably be something like PyWin32, is that correct? 

I really took this feature for granted but the more I look at it the more I realize it's not as simple and straight-forward as I expected. 

Thanks for the info.

**Comment by KeyWeeUsr** [comment]
Most likely, because as I was looking into SDL2 wiki, I found only [SDL_DropEvent](https://wiki.libsdl.org/SDL_DropEvent), which probably means there's no such thing as drag&dropping plain text in SDL2.
